### PR TITLE
test: make can-never-fail tests assert what their names claim

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

How to review this fast: commits are one concern each; 7 files, all under `*.Test` projects (verified: `git diff --name-only origin/master...HEAD | grep -v .Test` is empty); net -30 lines. No product code changed.

These tests passed unconditionally - assertions only inside `catch` blocks, `Assert.Pass()` endings, assertion-free bodies, or `async void` helpers whose assertions raced the test body. Each now asserts the thing its name claims, or is deleted where a retained test strictly subsumes it.

- `SpecialTransactionsTests`: drop the redundant trailing `Assert.Pass()` (real asserts precede it).
- `McsLockTests`: delete `SingleThreadAcquireRelease` (no asserts; subsumed by `ReacquireAfterReleaseSucceeds`), which now runs on a worker with a 10s bound so a broken release fails fast instead of hanging the run.
- `BloomFilterTests`: the concurrent add/read test now verifies every added hash is found afterwards (a bloom filter never returns a false negative, so a miss means a lost write). Capacity is 100k so saturation false-positives cannot mask a single lost write; misses are collected into one `Is.Empty` assertion.
- `FlatDbManagerPersistedTests`: `DisposeAsync_DisposesPersistedRepository` renamed to `DisposeAsync_CompletesPromptlyAndIsIdempotent` - `FlatDbManager.DisposeAsync` does not dispose the repository (the container owns it), so the old name asserted a contract that does not exist; it now asserts bounded worker-drain completion and an idempotent second dispose via `Assert.DoesNotThrowAsync`. The `Not.Null`-on-fresh-object constructor test is deleted.
- `NewPayloadHandlerRaceConditionTests`: two tests deleted. The memory-leak test asserted only inside `catch` and ended with `Assert.Pass`; the retained `..._when_timeout_happens_before_block_removed` test already covers the `finally`-block cleanup it aimed at, and iterating adds nothing (`_blockValidationTasks` is keyed by block hash, so resubmitting one block can never push the count above 1). The TrySet double-completion test was a strictly weaker duplicate of the concurrent-calls test above it (same scenario, but swallowing `OperationCanceledException` and keying `Assert.Fail` on exception message text).
- `BeaconHeadersSyncTests`: two `async void` helpers were invoked without awaiting - including one un-awaited call inside the other - so ~20 assertions raced the test body. Now `async Task` and awaited; all pass, so no latent failure was being masked.
- `DisconnectsAnalyzerTests`: all four tests had their assertions commented out as CI-flaky, leaving zero live assertions. Rewritten deterministically: reports are recorded while the constructor's 10s flush interval is in effect, and only then is the interval shortened, so a report can never race a flush. The clear test sends no second report at all (`Timer.Stop` does not drain a queued `Elapsed` callback, so no test-side quiesce is airtight); instead it relies on the analyzer's double-buffering - a lost clear resurfaces the stale count in later flushes, so the category must appear in exactly one flush across several more. `TestLogger.LogList` (a plain `List` written from the timer thread) is replaced by a `ConcurrentQueue` capture, and the previously leaked timer is disposed.

#### What types of changes does your code introduce?

- [x] Refactoring

#### Requires testing

- [x] Yes

#### If yes, did you write tests?

- [x] Yes

#### Notes on testing

- Mutation-checked: deliberate breaks (McsLock release no-op, BloomFilter.Add skipping bit writes, FlatDbManager dispose-guard bypass, NewPayloadHandler cleanup removal, DisconnectsAnalyzer clear removal, a perturbed BeaconHeaders expectation) each made the strengthened test fail, then were reverted. The clear-removal mutation was re-verified against the final single-report arrangement.
- Measured, not modelled: at the old `capacity: 10000` the bloom filter's false-positive rate (~34%) masked a single lost write one time in three; at `100_000` it is ~4e-5.
- Flake smoke: 0 failures across repeat runs (timing-sensitive fixtures 10-15x, others 3-6x, windows-x64 release).
- Full local suites green: Core.Test 5985, Merge.Plugin.Test, Network.Test, Xdc.Test. State.Flat.Test has 14 pre-existing `StorageLayerTests` failures on Windows only (identical on pristine master; `TearDown` deletes a directory while mmap views are live) - unrelated to this diff and invisible in CI, where the project runs on ubuntu.

#### Requires documentation update

- [ ] No

#### Requires explanation in Release Notes

- [ ] No

### CODE REVIEW

**Comment by benaadams** [comment]
@claude re-review

**Review by svlachakis — APPROVED** [review]

File: src/Nethermind/Nethermind.Network.Test/P2P/DisconnectsAnalyzerTests.cs [hunk 1]
```diff
@@ -14,61 +16,135 @@ namespace Nethermind.Network.Test.P2P
...
+                int flushesSeen = _logger.FlushCount;
+                Assert.That(PollUntil(() => _logger.FlushCount >= flushesSeen + 4), Is.True,
+                    "flush timer stopped ticking");
+                Assert.That(_logger.CountMatches(pattern), Is.EqualTo(times),
+                    () => $"a cleared category resurfaced in a later flush; got: {_logger.Dump()}");
+            }
+
+            public void Dispose() => DisconnectsAnalyzer.Dispose();
+
+            private static bool PollUntil(Func<bool> condition)
```
Comment (wurdum) [inline src/Nethermind/Nethermind.Network.Test/P2P/DisconnectsAnalyzerTests.cs]: I think we can use `SpinWait.SpinUntil` instead of the custom one

**Review by wurdum — APPROVED** [review]

**Review by svlachakis — APPROVED** [review]
